src/main/java/xyz/aadev/generitech/client/gui/power/GuiGenerator.java

```diff
@@ -0,0 +1,84 @@
+ */
+
+import net.darkhax.tesla.lib.TeslaUtils;
+import net.minecraft.entity.player.InventoryPlayer;
+import net.minecraft.inventory.Container;
+import org.lwjgl.util.Point;
+import org.lwjgl.util.Rectangle;
+import xyz.aadev.generitech.api.util.MachineTier;
+import xyz.aadev.generitech.client.gui.GuiBase;
+import xyz.aadev.generitech.common.container.machines.ContainerPulverizer;
+import xyz.aadev.generitech.common.container.power.ContanierGenerator;
+import xyz.aadev.generitech.common.tileentities.machines.TileEntityPulverizer;
+import xyz.aadev.generitech.common.tileentities.power.TileEntityPower;
+
+import java.util.ArrayList;
+
+public class GuiGenerator extends GuiBase {
+    private TileEntityPower tileEntity;
+    private MachineTier machineTier;
+    Rectangle powerBar;
+
+    public GuiGenerator(InventoryPlayer inventoryPlayer, TileEntityPower tileEntity) {
+        super(new ContanierGenerator(inventoryPlayer, tileEntity));
+        this.xSize = 176;
+        this.ySize = 166;
+        this.tileEntity = tileEntity;
+        this.machineTier = MachineTier.byMeta(tileEntity.getBlockMetadata());
+        powerBar = new Rectangle(98, 30, 14,28);
+    }
+
+    @Override
+    public void drawBG(int paramInt1, int paramInt2, int paramInt3, int paramInt4) {
+        bindTexture("gui/power/generator.png");
+        drawTexturedModalRect(paramInt1, paramInt2, 0, 0, this.xSize, this.ySize);
+
+
+
+        int powerLevel = (int) (tileEntity.powerStored() / 2000 + 1);
+        if (tileEntity.powerStored() > 48000) powerLevel = 25;
+        if (tileEntity.powerStored() == 0) powerLevel = -1;
+        int power = powerLevel - 25;
+        drawTexturedModalRect(paramInt1 + 100, paramInt2 + 31 - power, 176, 55 - powerLevel, 25, 25);
+
+        int fireOffset = tileEntity.getFuelOffset() + 1; // (x + 1) 1 and 11
+        drawTexturedModalRect(paramInt1 + 65, paramInt2 + 53 + fireOffset, 176, 1 + fireOffset, 14, 14 - fireOffset);
+
+
+
+    }
+
+    @Override
+    public void drawFG(int paramInt1, int paramInt2, int paramInt3, int paramInt4) {
+
+}
+
+
+    @Override
+    public void drawScreen(int mouse_x, int mouse_y, float btn){
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Rename this local variable name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.aadev.xyz/coding_rules#rule_key=squid%3AS00117)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Rename this local variable name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.aadev.xyz/coding_rules#rule_key=squid%3AS00117)